### PR TITLE
o [NXCM-3549] Add filtering of repo/group task setting combobox

### PR DESCRIPTION
**Pull request by nabcos**
Removal of repositories from the scheduled tasks UI was not supported. This change adds a filter based on the task id.

please check the filter expressions for sanity:
- DownloadIndexesTask : (repoType == 'proxy' && format == 'maven2') 
- OptimizeIndexTask : (format == 'maven2') 
- PublishIndexesTask : (format == 'maven2') 
- EvictUnusedProxiedItemsTask : (repoType == 'proxy')
- RebuildMavenMetadataTask : (format == 'maven2') 
- RepairIndexTask : (format == 'maven2') 
- UpdateIndexTask : (format == 'maven2') 
- SynchronizeShadowsTask : (repoType == 'virtual') 
- SnapshotRemoverTask : (format == 'maven2' && repoPolicy == 'SNAPSHOT')

### CODE REVIEW

**Comment by cstamas** [comment]
:beer:

File: nexus/nexus-webapp/src/main/webapp/js/repoServer/repoServer.SchedulesEditPanel.js [hunk 1]
```diff
@@ -176,12 +176,25 @@ Sonatype.repoServer.SchedulesEditPanel = function(config) {
...
+        {
+            name:'repoType'
+        },
+        {
+            name:'format'
+        },
+        {
+            name:'repoPolicy'
+        }
+    ]);
```
Comment (peterlynch) [inline nexus/nexus-webapp/src/main/webapp/js/repoServer/repoServer.SchedulesEditPanel.js:198]: does maven java formatting have to come to js too?

**Comment by peterlynch** [comment]
+1 like the idea and this solves immediate needs - am wondering if this filtering is handled on server side as well

File: nexus/nexus-webapp/src/main/webapp/js/repoServer/repoServer.SchedulesEditPanel.js (hunk 1 above)
Comment (nabcos) [inline nexus/nexus-webapp/src/main/webapp/js/repoServer/repoServer.SchedulesEditPanel.js:198]: Please define the format style for javascript.

Comment (nabcos) [inline nexus/nexus-webapp/src/main/webapp/js/repoServer/repoServer.SchedulesEditPanel.js:198]: Aside that IMO it's better than parts [like this](https://github.com/sonatype/nexus/blob/6d7582e521ffb328fc771a8ed7f770f5696b463c/nexus/nexus-webapp/src/main/webapp/js/repoServer/repoServer.SchedulesEditPanel.js#L1484) where there is no recognizable indention at all...

**Comment by nabcos** [comment]
@adreghiciu Agreed, but I'm not sure how to do that. IIUC using a 'remote' store might work that way, that (maybe ;) would query the repositories list resource with some parameters and get the filtered list directly from REST... but I think then we would have to mangle the list (add repo/group to name, add 'all repos' entry) also server-side... not sure how that would work.

(I did not investigate that because this was a one-beer-issue, and I was at about one and a half already because of ext js wrangling...)

**Comment by nabcos** [comment]
@peterlynch WDYM with server-side filtering? the tasks will check if they support a repository and do nothing if they're don't.

File: nexus/nexus-webapp/src/main/webapp/js/repoServer/repoServer.SchedulesEditPanel.js (hunk 1 above)
Comment (peterlynch) [inline nexus/nexus-webapp/src/main/webapp/js/repoServer/repoServer.SchedulesEditPanel.js:198]: no worries

**Comment by peterlynch** [comment]
@nabcos sounds like the server side is sanely handing dealing with the different repos. good enuff

**Pull request merged**